### PR TITLE
Clarify default referential actions for implicit many to many relationships

### PR DESCRIPTION
**Pull request by miguelff**
## Describe this PR

The default behavior for default referential actions (onUpdate, delete) on implicit many to many is undocumented, and the implementation inconsistent as manifested trough https://github.com/prisma/prisma/issues/16390 A fix for that bug is on its way, and should go together with this change in documentation.

## Changes

Clarified default referential actions for implicit many to many relationships 

## What issue does this fix?

None on its own, it partially addresses https://github.com/prisma/prisma/issues/16390 

## Any other relevant information

https://prismaio.notion.site/Informal-RFC-Implicit-many-to-many-0e16ef5755c747b2b9bf5496efc2050d

### CODE REVIEW

File: content/200-concepts/100-components/01-prisma-schema/06-relations/410-referential-actions/index.mdx [hunk 1]
```diff
@@ -116,10 +116,10 @@ Prisma supports the following referential actions:
...
 If you do not specify a referential action, Prisma uses the following defaults:
 
-| Clause     | Optional relations | Mandatory relations |
-| :--------- | :----------------- | :------------------ |
-| `onDelete` | `SetNull`          | `Restrict`          |
-| `onUpdate` | `Cascade`          | `Cascade`           |
+| Clause     | Optional relations | Mandatory relations | [Implicit Many-to-Many](#caveats) |
+| :--------- | :----------------- | :------------------ | :---------------------|
+| `onDelete` | `SetNull`          | `Restrict`          | `Cascade`             |
+| `onUpdate` | `Cascade`          | `Cascade`           | `Cascade`             |
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/410-referential-actions/index.mdx:122]: I am unsure: Is this addition helpful, considering that the table is introduced as "if you do not specify..." which one can not do for many-to-many?

File: content/200-concepts/100-components/01-prisma-schema/06-relations/410-referential-actions/index.mdx [hunk 2]
```diff
@@ -143,7 +143,7 @@ The schema does not explicitly define referential actions on the mandatory `auth
 
 The following caveats apply:
 
-- Referential actions are **not** supported on [implicit many-to-many relations](/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations). To use referential actions, you must define an explicit many-to-many relation and define your referential actions on the [join table](/concepts/components/prisma-schema/relations/troubleshooting-relations#how-to-use-a-relation-table-with-a-many-to-many-relationship).
+- Referential actions are **not** supported on [implicit many-to-many relations](/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations). To use referential actions, you must define an explicit many-to-many relation and define your referential actions on the [join table](/concepts/components/prisma-schema/relations/troubleshooting-relations#how-to-use-a-relation-table-with-a-many-to-many-relationship). The default referential actions for [implicit many-to-many relations](/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations) are `Cascade` both for `onDelete` and `onUpdate` foreign key constraints.
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/410-referential-actions/index.mdx:146]: Puh, this is hard to explain. My try:
```suggestion
- Referential actions are **not** supported on [implicit many-to-many relations](/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations). To use referential actions, you must define an explicit many-to-many relation and define your referential actions on the [join table](/concepts/components/prisma-schema/relations/troubleshooting-relations#how-to-use-a-relation-table-with-a-many-to-many-relationship). 
  - The implicit join table in [implicit many-to-many relations](/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations) uses `Cascade` for both referential actions `onDelete` and `onUpdate` for both relations. When one of the involved scalar relation fields is updated or deleted, the entry or row in the join table is also updated or deleted.
```

**Review by janpio — APPROVED** [review]
Generally fine, but left 2 comments. Feel free to consider.

**Review by andrew-walford-prisma — COMMENTED** [review]
LGTM, but I'm going to ask @keerlu to review, because this is a schema doc

File: content/200-concepts/100-components/01-prisma-schema/06-relations/410-referential-actions/index.mdx (hunk 2 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/410-referential-actions/index.mdx:146]: I like @janpio's suggestion... some wording tweak suggestions below:

```suggestion
- You can not specify the type of referential action on [implicit many-to-many relations](/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations). To specify the referential action, you must instead define an explicit many-to-many relation and define your referential actions on the [join table](/concepts/components/prisma-schema/relations/troubleshooting-relations#how-to-use-a-relation-table-with-a-many-to-many-relationship). 
  - The implicit join table in [implicit many-to-many relations](/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations) uses `Cascade` for both `onDelete` and `onUpdate` referential actions. When one of the involved scalar relation fields is updated or deleted, the entry or row in the join table is also updated or deleted.
```

File: content/200-concepts/100-components/01-prisma-schema/06-relations/410-referential-actions/index.mdx (hunk 1 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/410-referential-actions/index.mdx:122]: I also don't think this should be in the table in this way, as 'implicit many-to-many' is a different type of thing to 'optional' and 'mandatory'. Maybe a note linking to the other section?

```suggestion
| Clause     | Optional relations | Mandatory relations |
| :--------- | :----------------- | :------------------ | 
| `onDelete` | `SetNull`          | `Restrict`          |
| `onUpdate` | `Cascade`          | `Cascade`           |

Note that implicit many-to-many relations will always use `Cascade` for both `onUpdate` and `onDelete` referential actions. See [Caveats](#caveats) for more details.
```

**Review by keerlu — APPROVED** [review]

File: content/200-concepts/100-components/01-prisma-schema/06-relations/410-referential-actions/index.mdx (hunk 2 above)
Comment (janpio) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/410-referential-actions/index.mdx:146]: Like your iteration @keerlu

File: content/200-concepts/100-components/01-prisma-schema/06-relations/410-referential-actions/index.mdx (hunk 1 above)
Comment (janpio) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/410-referential-actions/index.mdx:122]: One problem, which I spent a lot of times to reflect in the suggested change below, is that the referential actions are _not_ on the m-n relation side itself - but invisible on the join table between them. Not sure if that is reflected in your suggestion @keerlu.

Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/410-referential-actions/index.mdx:122]: Yes, I'm just linking to the section below for the details.

**Comment by keerlu** [comment]
(This is currently on hold as the related bug fix has not gone in yet.)

**Comment by miguelff** [comment]
A fix should be paired with this change, and it's unfortunately not ready yet and won't be in this sprint I'm afraid. I'm leaning towards closing this and reopening it once the fix is ready. I will take care of updating the changes and reask for review. 

Thanks for your previous review!

**Pull request closed without merging**